packages/desktop-client/src/components/modals/EditRule.jsx

```diff
@@ -713,16 +742,46 @@ export default function EditRule({
   }
 
   function onRemoveAction(action) {
-    setActions(actions.filter(a => a !== action));
+    setActionSplits(splits =>
+      splits.map(actions => actions.filter(a => a !== action)),
+    );
+  }
+
+  function onRemoveSplit(splitIndexToRemove) {
+    setActionSplits(splits => {
+      const copy = [];
+      getActions(splits).forEach(action => {
+        const currentSplitIndex = action.options?.splitIndex ?? 0;
+        if (currentSplitIndex === splitIndexToRemove) {
+          return;
+        }
+        const newSplitIndex =
+          currentSplitIndex > splitIndexToRemove
+            ? currentSplitIndex - 1
+            : currentSplitIndex;
+        copy[newSplitIndex] = copy[newSplitIndex] ?? {
+          id: uuid(),
+          actions: [],
+        };
+        copy[newSplitIndex].actions.push({
+          ...action,
+          options: { ...action.options, splitIndex: newSplitIndex },
+        });
+      });
+      return copy;
+    });
   }
 
   function onApply() {
+    const selectedTransactions = transactions.filter(({ id }) =>
+      selectedInst.items.has(id),
+    );
     send('rule-apply-actions', {
-      transactionIds: [...selectedInst.items],
-      actions: actions.map(unparse),
+      transactions: selectedTransactions,
+      actions: actionSplits.flat().map(unparse),
```

Comment:
Oh, nice catch! that would be the case, but actually this is a bigger problem left over from when I did a small refactor last week. It should be `actions: getActions(actionSplits).map(unparse)` now since `actionSplits` is an array of objects. Updated!

Reply:
(side note: `getActions` can use `flatMap`, though)